services/venice-controller/src/main/java/com/linkedin/venice/controller/server/VeniceParentControllerStateHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.linkedin.venice.controller.server;
+
+import static com.linkedin.venice.ParentControllerState.ACTIVE;
+
+import com.linkedin.venice.HttpConstants;
+import com.linkedin.venice.ParentControllerState;
+import com.linkedin.venice.controller.Admin;
+import com.linkedin.venice.controllerapi.ControllerResponse;
+import com.linkedin.venice.exceptions.ErrorType;
+import org.apache.http.HttpStatus;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+
+/**
+ * Handler for checking ParentController state to handle requests.
+ */
+public class VeniceParentControllerStateHandler implements Route {
+  public static final String ACTIVE_CHECK_FAILURE_WARN_MESSAGE_PREFIX =
+      "Only active parent controllers are allowed to run ";
+  private Admin admin;
+  private Route route;
+
+  public VeniceParentControllerStateHandler(Admin admin, Route route) {
+    this.admin = admin;
+    this.route = route;
+  }
+
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    ParentControllerState parentControllerState = admin.getParentControllerState();
+    boolean isParent = admin.isParent();
+    if (parentControllerState != ACTIVE || !isParent) {
```

Comment:
This check is incorrect for child controllers and would reject all API request there. It should be 
```java
if (isParent && parentControllerState != ACTIVE) {
    ...
}
```